src/index.ts

```diff
@@ -20,22 +20,69 @@ import mapperReader from './mapperReader.js';
 import { MapperServer } from './mapperServer.js';
 import { BidiServerRunner } from './bidiServerRunner.js';
 import { IServer } from './iServer.js';
+import argparse from 'argparse';
+
+function parseArguments() {
+  var parser = new argparse.ArgumentParser({
+    add_help: true,
+    exit_on_error: false,
+  });
+
+  parser.add_argument('-p', '--port', {
+    help: 'Port that BiDi server should listen to. Default is 8080.',
+    type: 'int',
+    default: 8080,
+  });
+
+  parser.add_argument('-hl', '--headless', {
+    help: 'Sets if browser should run in headless or headful mode. Default is `--headless=true`.',
+    default: true,
+  });
+
+  parser.add_argument('-b', '--browser', {
+    help: 'Path to browser executable.',
+    required: true,
+  });
+
+  // `parse_known_args` puts known args in the first element of the result.
+  const args = parser.parse_known_args();
+  return args[0];
+}
 
 (async () => {
   try {
     console.log('Launching BiDi server.');
-    BidiServerRunner.run(_onNewBidiConnectionOpen, _onBidiConnectionClosed);
+
+    const args = parseArguments();
+
+    const bidiPort = args.port;
```

Comment:
This is where you'd have to add `PATH` support, but `args.port || process.env.PATH` won't work because `args.port` has a default. And `process.env.PATH || args.port` would make the environment variable take precedent over the argument, which is unusual and not great.

So just dropping it seems fine :)

Reply:
Done by tricking the default value.